proto/ai/traceable/agent/config/v1/config.proto

```diff
@@ -6,10 +6,19 @@ import "google/protobuf/wrappers.proto";
 
 option go_package = "github.com/Traceableai/agent-config/gen/go/v1";
 
-// Parent configuration object for Opa & Blocking Config
 message AgentConfig {
-    Opa opa = 1;
-    BlockingConfig blocking_config = 2;
+  Opa opa = 1 [deprecated = true];
+  BlockingConfig blocking_config = 2 [deprecated = true];
+  LibtraceableConfig libtraceable = 3;
```

Comment:
@davexroth, @tim-mwangi what do we think about this? The name `libtraceable` will appear though on config files. I need a new grouping for libtraceable because there are common libtraceable config such as `remote_config` and `debug_log` that affects various libtraceable functionality only i.e. `blocking`, `api_discovery` (api naming), and `sampling`.

Reply:
yeah, the problem is that this is a user facing config format, so we shouldn't expose out the details of libtraceable, and should instead focus on first class constructs such as block etc.